crates/kitsune_p2p/direct_test/src/lib.rs

```diff
@@ -0,0 +1,394 @@
+
+        for node in nodes.iter() {
+            for agent in node.local_agents.iter() {
+                let input = AgentHookInput {
+                    root: root.clone(),
+                    root_entry_hash: root_entry_hash.clone(),
+                    agent: agent.clone(),
+                    kdirect: node.kdirect.clone(),
+                };
+                if periodic_agent_hook(input).await.is_err() {
+                    break 'top;
+                }
+            }
+        }
+    }
+
+    Ok(())
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[tokio::test(flavor = "multi_thread")]
+    async fn sanity_run_for_five_seconds() {
+        init_tracing();
+
+        let mut config = KdTestConfig::default();
+        config.agent_init_hook = Box::new(|input| {
+            async move {
+                let AgentHookInput {
+                    root,
+                    root_entry_hash,
+                    agent,
+                    kdirect,
+                } = input;
+
+                let new_entry = KdEntryData {
+                    type_hint: "u.foo".to_string(),
+                    parent: root_entry_hash,
+                    author: agent.clone(),
+                    should_shard: true,
+                    reverify_interval_s: u32::MAX,
+                    verify: "".to_string(),
+                    data: serde_json::json!({
+                        "nonce": std::time::SystemTime::now()
+                            .duration_since(std::time::SystemTime::UNIX_EPOCH)
+                            .unwrap()
+                            .as_secs_f64(),
+                    }),
+                };
+                let new_entry = KdEntry::sign(&kdirect.get_persist(), new_entry).await?;
+                tracing::debug!(?new_entry);
+                kdirect.publish_entry(root, agent, new_entry).await?;
+
+                Ok(())
+            }
+            .boxed()
+        });
+
+        let test = KdTestHarness::start_test(config).await.unwrap();
+
+        tokio::time::sleep(std::time::Duration::from_secs(5)).await;
+
+        assert_eq!(2, test.nodes.len());
+        for node in test.nodes.iter() {
+            assert_eq!(2, node.local_agents.len());
+            for agent in node.local_agents.iter() {
+                let entries = node
+                    .kdirect
+                    .get_persist()
+                    .query_entries(
+                        test.root.clone(),
+                        agent.clone(),
+                        f32::MIN,
+                        f32::MAX,
+                        DhtArc::new(0, u32::MAX),
+                    )
+                    .await
+                    .unwrap();
+                let entry_count = entries.len();
+                let entry_hashes = entries.iter().map(|e| e.hash()).collect::<Vec<_>>();
+                tracing::info!(%entry_count, ?entry_hashes);
+                assert_eq!(5, entry_count);
```

Comment:
Why 5? Would be good to know why, in case this ever fails.

Reply:
:+1: added some comments